### ISSUE TITLE
Introduce Python classes

### ISSUE THREAD

**Issue by ethan-pierce** [csdms/ivy#111]
In several side conversations this week, students have indicated some confusion about how classes and instances work in Python. It might be worth taking some time to talk about how + why we define classes for each of our models/components and then developing the solver.py example into its own class.

**Comment by mdpiper** [comment]
Agree. We need to include at least a basic lesson on OOP that defines and explains concepts like:

* class
* instance
* method
* attribute
* constructor
* self variable

> [...] developing the solver.py example into its own class.

This is done in [diffusion.py](https://github.com/csdms/ivy/blob/main/lessons/python/ivy_diffusion/diffusion.py).

**Comment by mdpiper** [comment]
It might be nice to use the textbook example of `Shape`, `Circle`, `Rectangle`, and `Square`.

Make sure to connect that Landlab is OO, when you work with Landlab you're using OO features, and if you ever extend Landlab by writing a component, it'll be in the form (shape?) of a class.
